etw/cs/traceloggingdynamic/TraceLoggingDynamic.cs

```diff
@@ -674,13 +678,40 @@ private enum InType : byte
             Struct, // Use OutType for field count.
             CountedBinary, // counted, works for arrays, newer (not always supported)
             Mask = 31,
+            NullArray = 64, // Invalid type
+            UnicodeStringArray, // nul-terminated
+            AnsiStringArray, // nul-terminated
+            Int8Array,
+            UInt8Array,
+            Int16Array,
+            UInt16Array,
+            Int32Array,
+            UInt32Array,
+            Int64Array,
+            UInt64Array,
+            Float32Array,
+            Float64Array,
+            Bool32Array,
+            BinaryArray_NotSupported,
+            GuidArray, // size = 16
+            PointerArray_NotSupported,
+            FileTimeArray, // size = 8
+            SystemTimeArray, // size = 16
+            SidArray, // size = 8 + 4 * SubAuthorityCount
+            HexInt32Array,
+            HexInt64Array,
+            CountedStringArray, // counted
+            CountedAnsiStringArray, // counted
+            StructArray, // Use OutType for field count.
+            CountedBinaryArray, // counted, works for arrays, newer (not always supported)
         }
 
         private const byte InTypeCcount = 32;
         private const byte InTypeVcount = 64;
         private const byte ChainBit = 128;
 
         private string name;
+        private Encoding utf8NameEncoding;

```

Comment:
Note: Primary performance enhancement is to allow the user to control this instead of always using `Encoding.UTF8`.